### PR TITLE
New Python SDK for Cluster VErsion 6.6

### PR DESCRIPTION
**Pull request by naveena-maplelabs**
Changes:
- Implemented new APIs supported by cluster version 6.6.
- Added new modules and updated existing modules.

### CODE REVIEW

File: cohesity_management_sdk/controllers/active_directory_controller.py [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # -*- coding: utf-8 -*-
-# Copyright 2020 Cohesity Inc.
+# Copyright 2021 Cohesity Inc.
```
Comment (pyashish) [inline cohesity_management_sdk/controllers/active_directory_controller.py:3]: Good stuff changing the year.

File: cohesity_management_sdk/controllers/idps_controller.py [hunk 2]
```diff
@@ -19,7 +19,7 @@ def __init__(self, config=None, client=None, call_back=None):
         self.config = config
 
     def add_active_idp_principals(self):
-        """Does a POST request to /public/idp/principals.
+        """Does a POST request to /public/idps/principals.
```
Comment (pyashish) [inline cohesity_management_sdk/controllers/idps_controller.py:22]: was the a miss from previous release or the URL changed ?

File: cohesity_management_sdk/models/azure_protection_source.py [hunk 3]
```diff
@@ -197,17 +216,24 @@ def from_dictionary(cls,
         name = dictionary.get('name')
         num_cores = dictionary.get('numCores')
         physical_source_id = dictionary.get('physicalSourceId')
+        region = dictionary.get('region')
         resource_id = dictionary.get('resourceId')
         restore_task_id = dictionary.get('restoreTaskId')
         subscription_id = dictionary.get('subscriptionId')
         subscription_type = dictionary.get('subscriptionType')
+        tag_attributes = None
```
Comment (pyashish) [inline cohesity_management_sdk/models/azure_protection_source.py:224]: why is this handled differently than other variables ?

**Review by pyashish — APPROVED** [review]

File: cohesity_management_sdk/controllers/idps_controller.py (hunk 2 above)
Comment (naveena-maplelabs) [reply cohesity_management_sdk/controllers/idps_controller.py:22]: The URL has been modified.

File: cohesity_management_sdk/models/azure_protection_source.py (hunk 3 above)
Comment (naveena-maplelabs) [reply cohesity_management_sdk/models/azure_protection_source.py:224]: value for **tag_attributes** field is list of objects, so here we are iterating through the list of objects.
```
tag_attributes = None
if dictionary.get('tag_attributes')!= None:
            tag_attributes = list()
            for structure in dictionary.get('tag_attributes'):
                tag_attributes.append(cohesity_management_sdk.models.tag_attribute.TagAttribute.from_dictionary(structure))
```

**Comment by naveena-maplelabs** [comment]
@pyashish Can we have the datalock config changes in this release?

**Pull request merged**